src/adapters/voyage/helpers/embedding.ts

```diff
@@ -0,0 +1,25 @@
+import { VoyageAIClient } from "voyageai";
+import { Context } from "../../../types";
+import { SuperVoyage } from "./voyage";
+const VECTOR_SIZE = 1024;
+
+export class Embedding extends SuperVoyage {
+  protected context: Context;
+
+  constructor(client: VoyageAIClient, context: Context) {
+    super(client, context);
+    this.context = context;
+  }
+
+  async createEmbedding(text: string | null): Promise<number[]> {
+    if (text === null) {
+      return new Array(VECTOR_SIZE).fill(0);
+    } else {
+      const response = await this.client.embed({
+        input: text,
+        model: "voyage-large-2-instruct",
```

Comment:
My understanding is that the `-instruct` model we chose is better at taking instructions for reranking, but I didn't notice any instructions related to the embedding model.

Reply:
Be sure to consider the instructions from the [readme](https://github.com/voyage-ai/voyage-large-2-instruct). This seems non-intuitive but 

> For classification (or clustering) tasks, please set `input_type` to `None` and prepend "Classify the text: " (or "Cluster the text: ") to your texts. 

As I understand, you set `input_type` to `None` and then you can write custom instructions.